lib/chef/policy_builder/policyfile.rb

```diff
@@ -356,6 +354,30 @@ def config
         Chef::Config
       end
 
+      private
+
+      def compat_mode_manifest_for(cookbook_name, lock_data)
```

Comment:
There's a lot of duplicated code here

I think you can put the `Chef::Config[:policy_document_native_api]` logic in something like `cookbook_endpoint(cookbook_name, xyz_version)`